lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1492,6 +1503,41 @@ DIFile *SPIRVToLLVMDbgTran::getFile(const SPIRVId SourceId) {
                    getStringContinued(SourceArgs[TextIdx], Source));
 }
 
+std::tuple<uint64_t, const std::string>
+SPIRVToLLVMDbgTran::getBuildIdentifierAndStoragePath() {
+#ifndef NDEBUG
+  bool FoundBuildIdentifier{false};
+#endif
+  uint64_t BuildIdentifier = 0;
+  const std::string *StoragePathPtr{};
+
+  for (SPIRVExtInst *EI : BM->getDebugInstVec()) {
+    if (EI->getExtOp() == SPIRVDebug::BuildIdentifier) {
+      using namespace SPIRVDebug::Operand::BuildIdentifier;
+      SPIRVWordVec BuildIdentifierArgs = EI->getArguments();
+      assert(BuildIdentifierArgs.size() == OperandCount &&
+             "Invalid number of operands");
+      assert(!FoundBuildIdentifier &&
```

Comment:
FoundBuildIdentifier is local variable and it's seems to be always false.
I'd suggest to store pointer to BuildIdentifier as member of the class. And here we can check if its nullptr or not

Reply:
FoundBuildIdentifier is set when a BuildIdentifier instruction is found.  It is used to ensure that there is no more than one BuildIdentifier instruction.  Added BuildIdentifier and StoragePath members to class, so that this information can be re-used if necessary.  Adding these members also makes tuples unnecessary for returning values.